From cecae3b0646bdb554621f316b1ae3a74179cbc1d Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Tue, 12 Oct 2021 17:16:24 +0200
Subject: [PATCH 74/N] fixup! Work around old mingw-w64-headers in the i686
 flavor of MSYS2

This drops the workaround, which seems no longer to be needed (most
likely due to MSYS2 having decided to build i686 _MINGW_ packages
again).

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/cygwin/winlean.h | 43 -----------------------------------------
 winsup/utils/cygpath.cc |  3 ---
 winsup/utils/ps.cc      |  3 ---
 3 files changed, 49 deletions(-)

diff --git a/winsup/cygwin/winlean.h b/winsup/cygwin/winlean.h
index cda1b3e..3382686 100644
--- a/winsup/cygwin/winlean.h
+++ b/winsup/cygwin/winlean.h
@@ -104,49 +104,6 @@ extern "C" {
 #error "Version >= 8 of the w32api headers is required"
 #endif
 
-#ifdef __i386__ /* 32-bit MSYS2 won't get new enough mingw-w64-headers */
-
-/* Define extended memory API here as long as not available from mingw-w64. */
-
-typedef struct _MEM_ADDRESS_REQUIREMENTS
-{
-  PVOID LowestStartingAddress;
-  PVOID HighestEndingAddress;
-  SIZE_T Alignment;
-} MEM_ADDRESS_REQUIREMENTS, *PMEM_ADDRESS_REQUIREMENTS;
-
-typedef enum MEM_EXTENDED_PARAMETER_TYPE
-{
-  MemExtendedParameterInvalidType = 0,
-  MemExtendedParameterAddressRequirements,
-  MemExtendedParameterNumaNode,
-  MemExtendedParameterPartitionHandle,
-  MemExtendedParameterUserPhysicalHandle,
-  MemExtendedParameterAttributeFlags,
-  MemExtendedParameterMax
-} MEM_EXTENDED_PARAMETER_TYPE, *PMEM_EXTENDED_PARAMETER_TYPE;
-
-#define MEM_EXTENDED_PARAMETER_TYPE_BITS 8
-
-typedef struct DECLSPEC_ALIGN(8) MEM_EXTENDED_PARAMETER
-{
-  struct
-  {
-      DWORD64 Type : MEM_EXTENDED_PARAMETER_TYPE_BITS;
-      DWORD64 Reserved : 64 - MEM_EXTENDED_PARAMETER_TYPE_BITS;
-  };
-  union
-  {
-      DWORD64 ULong64;
-      PVOID Pointer;
-      SIZE_T Size;
-      HANDLE Handle;
-      DWORD ULong;
-  };
-} MEM_EXTENDED_PARAMETER, *PMEM_EXTENDED_PARAMETER;
-
-#endif
-
 /* VirtualAlloc2 is declared in <w32api/memoryapi.h> if NTDDI_VERSION
    >= NTDDI_WIN10_RS4 (a compile-time condition).  But we need the
    declaration unconditionally, even though the function will only be
diff --git a/winsup/utils/cygpath.cc b/winsup/utils/cygpath.cc
index 450be43..bc5f11d 100644
--- a/winsup/utils/cygpath.cc
+++ b/winsup/utils/cygpath.cc
@@ -24,9 +24,6 @@ details. */
 #define _WIN32_WINNT 0x0a00
 #define WINVER 0x0a00
 #define NOCOMATTRIBUTE
-#ifdef __i386__ /* 32-bit MSYS2 won't get new enough mingw-w64-headers */
-#define PMEM_EXTENDED_PARAMETER PVOID
-#endif
 #include <windows.h>
 #include <userenv.h>
 #include <shlobj.h>
diff --git a/winsup/utils/ps.cc b/winsup/utils/ps.cc
index fcfb395..478ed8e 100644
--- a/winsup/utils/ps.cc
+++ b/winsup/utils/ps.cc
@@ -6,9 +6,6 @@ This software is a copyrighted work licensed under the terms of the
 Cygwin license.  Please consult the file "CYGWIN_LICENSE" for
 details. */
 
-#ifdef __i386__ /* 32-bit MSYS2 won't get new enough mingw-w64-headers */
-#define PMEM_EXTENDED_PARAMETER PVOID
-#endif
 #include <errno.h>
 #include <stdio.h>
 #include <locale.h>
-- 
2.9.0

